[Tests] Unit-tests for ClipBoardManager and ConstantPropertiesView

real-darth opened the pull request:
**Make sure to Squash and Merge to fix a good commit message!**

This is a merge of 3 new unit-tests I have added to test new functionalities of the `ClipBoardManager.java` and `ConstantPropertiesView.java`. The test were developed independently as per requirement of the assignment.

Please review and see if anything seems strange or should be changed. For instance, both the tests have been placed in the same file because they were developed originally using similar variables. Maybe it makes more sense to separate the tests into two different files as this point? 

Some other tests have been worked on but will not be merged here since they are incomplete / do not work. The difficulty of implementing these test cases will be discussed with the group.

Related to #6

real-darth committed:
fix: added unit test and working on clipboard

Added unit test related to the storeSettings test. Working on a unit test that executes the clipboard functionality, but is proving to be very difficult

real-darth committed:
Added tests for ClipBoardManager (one now, and one for merge)

real-darth committed:
removed old implementation

real-darth committed:
refactor: moved test to new file, refactor and documentation

- Move test relating to ClipBoardManager.java to a separate file.
- Refactored tests in the new ClipBoardManagerTest file
- Added documentation to tests

real-darth committed:
fix: add unit test to ConstantsProperitesViewModel

Add test that fails currently because the solution is implemented in the main working branch. When this unit test is merged into the main branch it should work. If not create issue to solve it.

JXNCTED reviewed (dismissed):
Everything looks good

JXNCTED reviewed (changes requested):
It seems that the new test cases is raising some architecture test failures.

JXNCTED commented:
It is complaining about the use of accessing the package directly without annotation
```
java.lang.AssertionError: Architecture Violation [Priority: MEDIUM] - Rule 'no classes that are not annotated with @ApacheCommonsLang3Allowed should access classes that reside in a package 'org.apache.commons.lang3'' was violated (8 times):
```

Failed test
```
MainArchitectureTest. doNotUseApacheCommonsLang3
MainArchitectureTest. doNotUseJavaAWT
MainArchitectureTest. restrictStandardStreams
```

real-darth committed:
fix: removed library causing architecture error

changed using apache.commons.lang3 to clean string to instead use replaceAll argument with trim as well

real-darth committed:
fix: standardStream and JavaAWT architecture violations

- Added AllowAWT annotation to class when using the clipboard for testing
- Removed try-statements (causing standardStream architecture violation) and added throws to the methods instead

real-darth committed:
fix: add missing line of code
